test/extensions/common/wasm/foreign_test.cc

```diff
@@ -51,6 +52,48 @@ TEST_F(ForeignTest, ForeignFunctionEdgeCaseTest) {
   EXPECT_EQ(result, WasmResult::BadArgument);
 }
 
+TEST_F(ForeignTest, ForeignFunctionSetEnvoyFilterTest) {
+  Stats::IsolatedStoreImpl stats_store;
+  Api::ApiPtr api = Api::createApiForTest(stats_store);
+  Upstream::MockClusterManager cluster_manager;
+  Event::DispatcherPtr dispatcher(api->allocateDispatcher("wasm_test"));
+  auto scope = Stats::ScopeSharedPtr(stats_store.createScope("wasm."));
+  testing::NiceMock<LocalInfo::MockLocalInfo> local_info;
+
+  envoy::extensions::wasm::v3::PluginConfig plugin_config;
+  auto plugin = std::make_shared<Extensions::Common::Wasm::Plugin>(
+      plugin_config, envoy::config::core::v3::TrafficDirection::UNSPECIFIED, local_info, nullptr);
+  Wasm wasm(plugin->wasmConfig(), "", scope, *api, cluster_manager, *dispatcher);
+  proxy_wasm::current_context_ = &ctx_;
+
+  auto function = proxy_wasm::getForeignFunction("set_envoy_filter_state");
+  ASSERT_NE(function, nullptr);
+
+  auto result = function(wasm, "bad_arg", [](size_t size) { return malloc(size); });
+  EXPECT_EQ(result, WasmResult::BadArgument);
+
+  envoy::source::extensions::common::wasm::SetEnvoyFilterStateArguments args;
+  std::string in;
+
+  args.set_path("invalid.path");
+  args.set_value("unicorns");
+  args.SerializeToString(&in);
+  result = function(wasm, in, [](size_t size) { return malloc(size); });
+  EXPECT_EQ(result, WasmResult::NotFound);
+
+  args.set_path("envoy.tcp_proxy.cluster");
```

Comment:
You might need to link the filter state object library for this filter state.

Reply:
fixed the tests!